### PR TITLE
[wasm] JSImport feedback

### PR DESCRIPTION
**Pull request by pavelsavara**
Feedback from testing with @radekdoulik 
- fix exception marshaling on JSExport
- fix export class with no namespace on JSExport
- bind thisArg of function to closest scope in JSImport

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.mjs [hunk 1]
```diff
@@ -250,5 +263,14 @@ export function backback(arg1, arg2, arg3) {
 export const instance = {}
 
 globalThis.javaScriptTestHelper = instance;
+globalThis.data = new JSData("i-n-s-t-a-n-c-e");
+globalThis.rebound = {
+    // our JSImport will try to bind it to `globalThis.rebound` 
+    // but it would stay bound to globalThis.data
+    // because once the function is bound, it would stay bound to the first object and can't be re-bound subsequently
```
Comment (kg) [inline src/libraries/System.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.mjs:270]: interesting that bind works this way, but it makes sense

File: src/mono/wasm/runtime/invoke-cs.ts [hunk 2]
```diff
@@ -181,13 +183,15 @@ function _walk_exports_to_set_function(assembly: string, namespace: string, clas
     scope = assemblyScope;
     for (let i = 0; i < parts.length; i++) {
         const part = parts[i];
-        let newscope = scope[part];
-        if (!newscope) {
-            newscope = {};
-            scope[part] = newscope;
+        if (part != "") {
+            let newscope = scope[part];
+            if (!newscope) {
```
Comment (kg) [inline src/mono/wasm/runtime/invoke-cs.ts:188]: explicit undefined check as above

**Review by kg — APPROVED** [review]

**Review by radekdoulik — APPROVED** [review]

**Review by maraf — APPROVED** [review]

**Comment by pavelsavara** [comment]
`Not found: args[0] = x` is https://github.com/dotnet/runtime/issues/71887

**Pull request merged**